Pass several privilege tests and fix RV32I comment

Risheng1128 opened the pull request:
This commit does the following things:
- Pass serveral privilege instruction and exceptions tests like ebreak, 
  misaligned load word and misaligned store word.
- Fix the comment of different type of RV32I instructions  format like 
  I, R and J-type and so on.

Because the compiler option `rv32gc` will make `ebreak` be compiled
as `c.ebreak` format, change it to `rv32g`.

jserv reviewed (changes requested):
Update README.md in top-level directory.

jserv commented:
I would anticipate the commit being divided into two:
1. Refine comments for RV32 instruction descriptions.
2. Improve compliance for privileged instructions.

Risheng1128 committed:
Refine comments for RV32 instruction descriptions

Fix the comment for different types of RV32I instructions format
like I, R and J-type and so on.

Risheng1128 committed:
Improve compliance for privileged instructions

Pass serveral privilege instruction and exceptions tests like ebreak,
misaligned load word and misaligned store word.

Moreover, the privilege instruction ecall doesn't pass the test
because it only dispatches the syscall number to corresponding handler.

Therefore, it needs to implement the exception handler with control
and status registers (CSRs).

Risheng1128 commented:
The current result of privilege tests:
```shell
Check ebreak                    ... OK 
Check ecall                     ... FAIL 
Check misalign1-jalr-01         ... OK 
Check misalign2-jalr-01         ... OK 
Check misalign-beq-01           ... OK 
Check misalign-bge-01           ... OK 
Check misalign-bgeu-01          ... OK 
Check misalign-blt-01           ... OK 
Check misalign-bltu-01          ... OK 
Check misalign-bne-01           ... OK 
Check misalign-jal-01           ... OK 
Check misalign-lh-01            ... OK 
Check misalign-lhu-01           ... OK 
Check misalign-lw-01            ... OK 
Check misalign-sh-01            ... OK 
Check misalign-sw-01            ... OK 
```

jserv reviewed (changes requested):
Analyze the reason why ecall fails and describe in the git commit message.

Risheng1128 commented:
The privilege instruction `ecall` doesn't pass the test because it only
dispatches the syscall number to corresponding handler in current
rv32emu.
Therefore, it needs to implement the exception handler with control
and status registers (CSRs).

jserv commented:
Thank @Risheng1128 for contributing!
